src/cpp/core/include/core/system/ProcessArgs.hpp

```diff
@@ -81,8 +81,8 @@ class ProcessArgs : boost::noncopyable
       {
          for (std::size_t i = 0; i<argCount_; ++i)
             delete [] args_[i] ;
-         delete [] args_ ;
       }
+      delete [] args_ ;
```

Comment:
You actually can delete a NULL pointer, see
http://www.cplusplus.com/reference/new/operator%20delete[]/

On Wed, Feb 24, 2016 at 1:24 PM, Kevin Ushey <notifications@github.com>
wrote:

> In src/cpp/core/include/core/system/ProcessArgs.hpp
> <https://github.com/rstudio/rstudio/pull/529#discussion_r53982311>:
>
> >        }
> > +      delete [] args_ ;
>
> This doesn't look right either -- the constructor calls setArgs(), which
> also calls freeArgs(), which means we might attempt to delete a NULL
> pointer. We should guard the deletion.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/rstudio/rstudio/pull/529/files#r53982311>.
>